semgrep_metrics.atd

```diff
@@ -60,6 +60,7 @@ type environment = {
     ?rulesHash <ocaml mutable>: sha256 option;
     (* TODO: should be just boolean *)
     ci <ocaml mutable>: string nullable;
+    ?baselineCommitHash <ocaml mutable> : sha256 option;
```

Comment:
when adding  new fields, please add a comment before to explain when it was introduced like
(* since semgrep 1.46 *)

Reply:
also maybe add a small comment before explaining what it is, in which context this is set.
Most of the other fields do not have comments because this .atd was written first by looking at the code in metrics_.py which didn't have much comment, but for new fields we should try to be better.

Reply:
Done.